### PR TITLE
Rewrite webhooks / Add Client

### PR DESCRIPTION
**Pull request by domenkozar**
Tried my best to split into commits, but included all the cabal and readme changes in the very last one.

### CODE REVIEW

File: src/FastSpring/Client.hs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import           Servant.Client.Generic
+
+import FastSpring.Client.API (fastSpringAPI, FastSpringAPI(..))
+
+
+-- TODO: mkClient :: BasicAuthData -> ...
+client :: FastSpringAPI (AsClientT ClientM)
+client = fromServant $ Servant.Client.client fastSpringAPI
+
+mkClient :: ClientM a -> IO (Either ServantError a)
```
Comment (roberth) [inline src/FastSpring/Client.hs]: `runClient manager cmd`

File: src/FastSpring/UnixTimeMili.hs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  ( UnixTimeMili(..)
+  ) where
+
+import Data.Aeson              ( FromJSON(..), withScientific )
+import Data.Time.Clock         ( UTCTime )
+import Data.Time.Clock.System  ( systemToUTCTime, SystemTime(..) )
+
+
+-- represents unix time in miliseconds
+newtype UnixTimeMili =
```
Comment (roberth) [inline src/FastSpring/UnixTimeMili.hs]: `UnixTimeMillis`, `UnixTimeMilliSeconds`, or at least spell with two `l`'s

Comment (roberth) [reply src/FastSpring/UnixTimeMili.hs]: Also comment should be haddock and point out that it's for serialization purposes

File: src/FastSpring/WebHook/Event.hs [hunk 3]
```diff
@@ -57,8 +63,8 @@ data EventParsingError
 parse :: Event -> Either EventParsingError EventData
 parse event =
   case _type event of
-      "subscription.deactivated" -> SubscriptionDeactivated <$> parseData
-      "subscription.activated" -> SubscriptionActivated <$> parseData
+      --"subscription.deactivated" -> SubscriptionDeactivated <$> parseData
```
Comment (roberth) [inline src/FastSpring/WebHook/Event.hs:67]: umm

File: src/FastSpring/WebHook/Signature.hs [hunk 4]
```diff
@@ -1,23 +1,34 @@
...
 import Data.ByteString (ByteString)
+import GHC.Generics (Generic)
 import Crypto.MAC.HMAC (hmac, HMAC)
 import Crypto.Hash.Algorithms (SHA256)
 import Servant.API
 
-import FastSpring.Settings (Settings(..))
+
+newtype SignatureSecret = SignatureSecret
+  { unwrapSecret :: ByteString
```
Comment (roberth) [inline src/FastSpring/WebHook/Signature.hs]: `SignatureSecret { unwrapSecret = ... }` is weird unless in pattern syntax. You can use `coerce` to unwrap, so you might as well use positional syntax.

File: src/FastSpring/WebHook/Event.hs (hunk 3 above)
Comment (domenkozar) [reply src/FastSpring/WebHook/Event.hs:67]: Missing todo :)

**Comment by domenkozar** [comment]
@roberth I've addressed your comments.

File: src/FastSpring/WebHook/Signature.hs [hunk 5]
```diff
@@ -1,34 +1,42 @@
...
 
-newtype Signature = Signature
-  { unwrap :: ByteString }
-  deriving (Show)
+newtype SignatureSecret = SignatureSecret ByteString
+
+newtype Signature = Signature ByteString
+  deriving (Show, Generic)
+
+-- useful for logging
```
Comment (roberth) [inline src/FastSpring/WebHook/Signature.hs]: Instances can have haddock. I am assuming this is not part of the FastSpring API? How is it rendered? (all non-Generic *JSON instances should document that)

File: src/FastSpring/WebHook/Signature.hs [hunk 6]
```diff
@@ -1,34 +1,42 @@
...
+instance ToJSON Signature where
+  toJSON (Signature bs) = toJSON (toSL bs :: Text)
 
 instance Eq Signature where
-  l == r = unwrap l `constEq ` unwrap r
+  (Signature l) == (Signature r) = l `constEq ` r
```
Comment (roberth) [inline src/FastSpring/WebHook/Signature.hs:34]: :heart: but please share this with the user via haddock

**Review by roberth — CHANGES_REQUESTED** [review]

**Comment by domenkozar** [comment]
@roberth second round, added haddock docs.

**Review by roberth — APPROVED** [review]

**Pull request merged**